### PR TITLE
Fix invalid invocations of errors.LineTooLong.

### PR DESCRIPTION
**Pull request by filmor**
## What do these changes do?

Fix unhelpful messages on `LineTooLong` exceptions due to invalid or incomplete calls to their constructor.

## Are there changes in behavior for the user?

Only in the message that they see in this case.

## Checklist
- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] Add yourself to `CONTRIBUTORS.txt`
  - The format is &lt;Name&gt; &lt;Surname&gt;.
  - Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new entry to `CHANGES.rst`
  - Choose any open position to avoid merge conflicts with other PRs.
  - Add a link to the issue you are fixing (if any) using `#issue_number` format at the end of changelog message. Use Pull Request number if there are no issues for PR or PR covers the issue only partially.

### CODE REVIEW

File: aiohttp/protocol.py [hunk 1]
```diff
@@ -102,7 +102,8 @@ def parse_headers(self, lines):
                     header_length += len(line)
                     if header_length > self.max_field_size:
                         raise errors.LineTooLong(
-                            'limit request headers fields size')
+                            'request header field %s' % bname.decode("utf8"),
```
Comment (popravich) [inline aiohttp/protocol.py]: `bname.decode('utf-8')` can fail with own exception

File: aiohttp/protocol.py [hunk 2]
```diff
@@ -113,7 +114,8 @@ def parse_headers(self, lines):
             else:
                 if header_length > self.max_field_size:
                     raise errors.LineTooLong(
-                        'limit request headers fields size')
+                        'request header field %s' % bname.decode("utf8"),
```
Comment (popravich) [inline aiohttp/protocol.py]: same as above

File: aiohttp/protocol.py [hunk 3]
```diff
@@ -173,7 +175,7 @@ def __call__(self, out, buf):
             raw_data = yield from buf.readuntil(
                 b'\r\n\r\n', self.max_headers)
         except errors.LineLimitExceededParserError as exc:
-            raise errors.LineTooLong(exc.limit) from None
+            raise errors.LineTooLong('request header', exc.limit) from None
```
Comment (popravich) [inline aiohttp/protocol.py:180]: this looks inconvenient: `LineTooLong` above is raised with single argument (formated string) and with two args here

Comment (filmor) [reply aiohttp/protocol.py:180]: It's not, and the call here is wrong. Check out the constructor of `LineTooLong` (didn't change that one), it expects a "reason" and a limit that was exceeded. The way this was passed before resulted in a broken error message like `"400 got more than Unknown bytes when reading 32768"` when it should have read `"400 got more than 32768 when reading request header"`.

**Comment by filmor** [comment]
I'll fix the failing tests and add some more.

File: aiohttp/protocol.py (hunk 3 above)
Comment (popravich) [reply aiohttp/protocol.py:180]: true, missed that one.

File: aiohttp/protocol.py (hunk 1 above)
Comment (filmor) [reply aiohttp/protocol.py]: Can it though? This is caught before by "Invalid Header". Changed it anyhow :)

**Comment by asvetlov** [comment]
Thanks!

**Pull request merged**